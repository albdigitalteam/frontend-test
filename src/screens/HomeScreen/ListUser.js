import React, { memo } from 'react';
import { FlatList } from 'react-native';
import styled from 'styled-components/native';

import PropTypes from 'prop-types';

import Label from '@/components/Label';
import colors from '@/theme/colors';

const ListUser = ({ users, isLoading }) => {
  const imageUsers = {
    1: require('@/assets/images/1.png'),
    2: require('@/assets/images/2.png'),
    3: require('@/assets/images/3.png'),
    4: require('@/assets/images/4.png'),
    5: require('@/assets/images/5.png'),
    6: require('@/assets/images/6.png'),
    7: require('@/assets/images/7.png'),
    8: require('@/assets/images/8.png'),
    9: require('@/assets/images/9.png'),
    10: require('@/assets/images/10.png')
  };
  const renderItem = ({ item: { id, name } }) => {
    return (
      <>
        <StyledViewImage>
          <StyledImage source={imageUsers[id]} />
          <Label
            width={85}
            numberOfLines={1}
            fontSize={12}
            marginLeft={10}
            textAlign='center'
            lineHeight={24}
          >
            {name}
          </Label>
        </StyledViewImage>
      </>
    );
  };

  return (
    <StyledContentUsers>
      <FlatList
        data={users}
        horizontal
        keyExtractor={item => item.id}
        showsHorizontalScrollIndicator={false}
        renderItem={renderItem}
        isLoading={isLoading}
      />
    </StyledContentUsers>
  );
};

const StyledContentUsers = styled.View`
  width: 100%;
  height: 100px;
  align-items: center;
  justify-content: center;
  padding: 8px;
  background-color: ${colors.COLOR_WHITE};
`;

const StyledImage = styled.Image`
  width: 60px;
  height: 60px;
  border-radius: 50px;
  margin-left: 8px;
`;

const StyledViewImage = styled.View`
  flex: 1;
  align-items: center;
  justify-content: center;
`;

ListUser.defaultProps = {
  users: [],
  isLoading: false
};

ListUser.propTypes = {
  isLoading: PropTypes.bool,
  users: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
      username: PropTypes.string,
      email: PropTypes.string
    })
  )
};

export default memo(ListUser);
