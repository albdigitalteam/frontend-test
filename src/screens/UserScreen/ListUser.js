import React, { memo, useCallback } from 'react';
import { FlatList, RefreshControl } from 'react-native';
import styled from 'styled-components/native';

import PropTypes from 'prop-types';

import colors from '@/theme/colors';

import CardUser from '@/components/CardUser';

const ListUser = ({ users, isLoading, onRefresh }) => {
  const renderItem = ({ item: { id, name } }) => {
    return <Card id={id} name={name} />;
  };

  const reflashControl = useCallback(
    () => (
      <RefreshControl
        progressViewOffset={32}
        tintColor={colors.PRIMARY}
        colors={[colors.PRIMARY]}
        refreshing={isLoading}
        onRefresh={onRefresh}
      />
    ),
    [isLoading, onRefresh]
  );

  return (
    <StyledContainer>
      <FlatList
        data={users}
        keyExtractor={item => item.id}
        showsVerticalScrollIndicator={false}
        refreshControl={reflashControl()}
        renderItem={renderItem}
        refreshing={isLoading}
      />
    </StyledContainer>
  );
};

const StyledContainer = styled.View`
  flex: 1;
  background-color: ${colors.COLOR_GRAY};
`;

const Card = styled(CardUser)`
  margin-top: 10px;
`;
ListUser.defaultProps = {
  users: [],
  isLoading: false,
  onRefresh: () => {}
};

ListUser.propTypes = {
  isLoading: PropTypes.bool,
  users: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
      username: PropTypes.string,
      email: PropTypes.string
    })
  ),
  onRefresh: PropTypes.func
};

export default memo(ListUser);
