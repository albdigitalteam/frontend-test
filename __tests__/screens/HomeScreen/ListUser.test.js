import 'react-native';
import React from 'react';
import {
  matchSnapshotWithTheme,
  renderWithTheme
} from '@/helpers/functions/testing';

import ListUser from '@/screens/HomeScreen/ListUser';

describe('tests in ListUser Screen', () => {
  test('matches snapshot', () => {
    matchSnapshotWithTheme(<ListUser />);
  });

  test('renders correctly', () => {
    renderWithTheme(<ListUser />);
  });

  test('ListUser renders correctly', () => {
    <ListUser users={[]} isLoading={false} />;
  });
});
