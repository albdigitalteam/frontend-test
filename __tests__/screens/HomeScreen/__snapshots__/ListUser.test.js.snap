// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in ListUser Screen matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "height": 100,
        "justifyContent": "center",
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
        "width": "100%",
      },
    ]
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    isLoading={false}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
